service/Microsoft.Azure.Devices/ServiceClient.cs

```diff
@@ -44,6 +45,26 @@ public static ServiceClient CreateFromConnectionString(string connectionString)
             return CreateFromConnectionString(connectionString, TransportType.Amqp);
         }
 
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases unmanaged and - optionally - managed resources.
+        /// </summary>
+        /// <param name="disposing"><c>true</c> to release both managed and unmanaged resources; <c>false</c> to release only unmanaged resources.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if ( disposing )
+            {
+                // Call CloseAsync() synchronously avoiding AggregateException.
+                this.CloseAsync().GetAwaiter().GetResult();
+            }
+        }
```

Comment:
Since this is an abstract class, you can declare it as protected abstract void Dispose(bool disposing);

---
In reply to: [103077630](https://github.com/Azure/azure-iot-sdk-csharp/pull/57#discussion_r103077630) [](ancestors = 103077630)